packages/ui/v2/core/apps/AllApps.tsx

```diff
@@ -122,7 +122,13 @@ export default function AllApps({ apps }: AllAppsPropsType) {
       </div>
       <div className="grid grid-cols-2 gap-3 md:grid-cols-4" ref={appsContainerRef}>
         {apps
-          .filter((app) => (selectedCategory !== null ? app.category === selectedCategory : true))
+          .filter((app) =>
+            selectedCategory !== null
+              ? app.categories
+                ? app.categories.includes(selectedCategory)
+                : app.category === selectedCategory
```

Comment:
Just in case there are apps with `category` and not a list of `categories`. But the main issue regarding app count not correct was fixed by filtering existent apps from app registry, not taking into consideration all categories in the database.